### PR TITLE
New recipient fields

### PR DESCRIPTION
**Pull request by achen2401**
https://www.pivotaltracker.com/story/show/184772062
The follow new editable fields are added:

- ISACC user ID (e.g., unique ID that is not PHI)
- Preferred name
- Pronouns
- Multiple emergency contacts
- Address (for 911)
- "Study start date"
- "ISAAC status", drop-down / mutually exclusive:
Active
Completed
On Hold
Lost
Refused further CC
Withdrawn by PI
Deceased
- "Suicide risk level" (e.g., C-SSRS score & PHQ9 score)

Existing fields that are now made editable:
- First Name
- Last Name
- Gender
- DOB

**Note**, in the story, this field is requested `"Account created" - auto-populate at patient creation.` I am not currently finding a reliable field to indicate this, so didn't add. 

Saved resource example:
- Patient resource to include those new fields
```
{
  "resourceType": "Patient",
  "extension": [
    {
      "url": "isacc.app/study-start-date",
      "valueDate": "2023-06-21"
    },
    {
      "url": "isacc.app/study-status",
      "valueString": "active"
    },
    {
      "url": "http://hl7.org/fhir/StructureDefinition/individual-genderIdentity",
      "valueString": "him/himself"
    }
  ],
  "identifier": [
    {
      "system": "http://isacc.app/user-id",
      "value": "ISACC-123"
    }
  ],
  "name": [
    {
      "family": "Abbott",
      "given": [
        "Barney"
      ]
    },
    {
      "use": "usual",
      "given": [
        "Barne"
      ]
    }
  ],
  "telecom": [
    {
      "system": "phone",
      "value": "555-901-9296",
      "use": "home"
    },
    {
      "system": "sms",
      "value": "6128121862"
    }
  ],
  "gender": "male",
  "birthDate": "1964-03-18",
  "deceasedDateTime": "1989-02-15T08:54:18+00:00",
  "address": [
    {
      "use": "home",
      "text": "195 Binney ST"
    }
  ],
  "contact": [
    {
      "relationship": [
        {
          "coding": [
            {
              "system": "http://hl7.org/fhir/ValueSet/patient-contactrelationship",
              "code": "C",
              "display": "Emergency Contact"
            }
          ]
        }
      ],
      "name": {
        "text": "Amy Contact"
      },
      "telecom": [
        {
          "system": "phone",
          "value": "2222222222"
        },
        {
          "system": "email",
          "value": "achen2401+test@gmail.com"
        }
      ]
    },
   {
      "relationship": [
        {
          "coding": [
            {
              "system": "http://hl7.org/fhir/ValueSet/patient-contactrelationship",
              "code": "C",
              "display": "Emergency Contact"
            }
          ]
        }
      ],
      "name": {
        "text": "Test Test"
      },
      "telecom": [
        {
          "system": "phone",
          "value": "1231231111"
        },
        {
          "system": "email",
          "value": "achen2401+test3@gmail.com"
        }
      ]
    }
      ],
      "name": {
        "text": "Test Contact"
      },
      "telecom": [
        {
          "system": "phone",
          "value": "1111111"
        },
        {
          "system": "email",
          "value": "achen2401+test2@gmail.com"
        }
      ]
    }
  ],
  "generalPractitioner": [
    {
      "reference": "Practitioner/792eb592-6944-4dad-87c8-66db4f237382",
      "type": "Practitioner"
    }
  ]
}
```
- PHQ9 score (Observation):
```
{
        "resourceType": "Observation",
        "id": "1899524",
        "meta": {
          "versionId": "2",
          "lastUpdated": "2023-06-27T15:36:37.168-04:00"
        },
        "code": {
          "coding": [
            {
              "system": "http://loinc.org",
              "code": "44261-6",
              "display": "PHQ9 score"
            }
          ]
        },
        "subject": {
          "reference": "Patient/5ee05359-57bf-4cee-8e89-91382c07e162"
        },
        "effectiveDateTime": "2023-06-27T19:29:23.642Z",
        "valueQuantity": {
          "value": 24
        }
      }
```
- C-SSRS score
```
{
        "resourceType": "Observation",
        "id": "1899523",
        "meta": {
          "versionId": "2",
          "lastUpdated": "2023-06-27T16:37:34.054-04:00"
        },
        "code": {
          "coding": [
            {
              "system": "http://loinc.org",
              "code": "93373-9",
              "display": "C-SSRS score"
            }
          ]
        },
        "subject": {
          "reference": "Patient/5ee05359-57bf-4cee-8e89-91382c07e162"
        },
        "effectiveDateTime": "2023-06-27T19:29:27.390Z",
        "valueQuantity": {
          "value": 23
        }
      }
```

screenshots:
Screen Shot 2023-06-29 at 12 56 39 PM

Screen Shot 2023-06-27 at 2 18 54 PM
Screen Shot 2023-06-27 at 2 19 13 PM

### CODE REVIEW

File: src/model/Patient.ts [hunk 1]
```diff
@@ -118,43 +166,174 @@ export default class Patient implements IPatient {
...
-        this.address.find((a) => a.use !== "home");
+    let address: IAddress = this.address.find((a) => a.use !== "home");
     if (address == null) {
       address = {} as IAddress;
       this.address.push(address);
     }
     address.postalCode = text;
   }
+
+  addContact(
```
Comment (mcjustin) [inline src/model/Patient.ts]: To disambiguate, can you please name this and related functions to `*EmergencyContact`, eg "addEmergencyContact"

File: src/model/Patient.ts [hunk 2]
```diff
@@ -118,43 +166,174 @@ export default class Patient implements IPatient {
...
+      });
+    }
+    if (email) {
+      if (!contactToAdd.telecom) contactToAdd.telecom = [];
+      contactToAdd.telecom.push({
+        system: ContactPointSystemKind._email,
+        value: email,
+      });
+    }
+    this.contact.push(contactToAdd);
```
Comment (mcjustin) [inline src/model/Patient.ts:255]: Can you please also set contact.relationship to `EP` per https://www.hl7.org/fhir/valueset-patient-contactrelationship.html ?

**Review by mcjustin — CHANGES_REQUESTED** [review]

File: src/model/Patient.ts (hunk 2 above)
Comment (mcjustin) [reply src/model/Patient.ts:255]: As I follow the breadcrumbs from my other link to https://terminology.hl7.org/5.1.0/CodeSystem-v2-0131.html#v2-0131-EP, I think `coding` should be `C` (`EC` deprecated) and probably best to include `display` of `Emergency Contact` for transparency. Does that make sense, or am I overlooking something you found?

Comment (achen2401) [reply src/model/Patient.ts:255]: thanks @mcjustin I did make the fix, e.g.
```
{
      "relationship": [
        {
          "coding": [
            {
              "system": "http://hl7.org/fhir/ValueSet/patient-contactrelationship",
              "code": "C",
              "display": "Emergency Contact"
            }
          ]
        }
      ],
      "name": {
        "text": "Test Test"
      },
      "telecom": [
        {
          "system": "phone",
          "value": "1231231111"
        },
        {
          "system": "email",
          "value": "achen2401+test3@gmail.com"
        }
      ]
    }
```

**Review by mcjustin — APPROVED** [review]

**Pull request merged**